### PR TITLE
fix setAPIVersion on native side

### PR DESCRIPTION
**Pull request by ex3ndr**
Wrapping setAPIVersion in native side for compatibility with jest.

### CODE REVIEW

**Comment by ex3ndr** [comment]
Sorry for this mess, but might work now. Not sure how to test this properly since in my setup issue is very flaky.

File: src/napi/module.cpp [hunk 1]
```diff
@@ -55,8 +56,16 @@ static napi_value setAPIVersion(napi_env env, napi_callback_info info) {
 
   int32_t apiVersion;
   NAPI_OK_OR_RETURN_NULL(env, napi_get_value_int32(env, args[0], &apiVersion));
-
-  FDB_OK_OR_RETURN_NULL(env, fdb_select_api_version(apiVersion));
+  
+  if (previousApiVersion != 0) {
+    if (apiVersion != previousApiVersion) {
+      napi_throw_error(env, NULL, "foundationdb already initialized with another API version");
```
Comment (josephg) [inline src/napi/module.cpp]: This is missing the correct foundationdb error code. And by supplying that error code, the string shouldn't be necessary.

I think it'll be easier to just call `fdb_select_api_version` if the version doesn't match, and let that function error back to the user in the standard way.

File: src/napi/module.cpp [hunk 2]
```diff
@@ -45,6 +45,7 @@ using namespace std;
 static uv_thread_t fdbThread;
 
 static bool networkStarted = false;
+static int32_t previousApiVersion = 0;
```
Comment (josephg) [inline src/napi/module.cpp]: I'm wondering if you should add header version here too - it feels cleaner to do so, but I'm not sure its actually necessary.

File: src/napi/module.cpp [hunk 3]
```diff
@@ -67,8 +76,16 @@ static napi_value setAPIVersionImpl(napi_env env, napi_callback_info info) {
   NAPI_OK_OR_RETURN_NULL(env, napi_get_value_int32(env, args[0], &apiVersion));
   int32_t headerVersion;
   NAPI_OK_OR_RETURN_NULL(env, napi_get_value_int32(env, args[1], &headerVersion));
-
-  FDB_OK_OR_RETURN_NULL(env, fdb_select_api_version_impl(apiVersion, headerVersion));
+  
+  if (previousApiVersion != 0) {
+    if (apiVersion != previousApiVersion) {
+      napi_throw_error(env, NULL, "foundationdb already initialized with another API version");
```
Comment (josephg) [inline src/napi/module.cpp]: ... ditto here. If the versions don't match just let it call `fdb_select_api_version_impl` and let that throw.

Comment (ex3ndr) [reply src/napi/module.cpp]: Semantics are different. fdb library doesn't allow you to call two times at all while we are allowing to call it several times with same apiVersion.

File: src/napi/module.cpp (hunk 2 above)
Comment (ex3ndr) [reply src/napi/module.cpp]: Well, i just replicated same code from typescript. I am not sure either.

Comment (josephg) [reply src/napi/module.cpp]: 🤷‍♀️ mmm lets leave it for now...

File: src/napi/module.cpp (hunk 3 above)
Comment (josephg) [reply src/napi/module.cpp]: Yes. I'm trying to take advantage of that so we return a canonical FDB error to the user if they try and set the version twice with different version numbers.

Alternately use `wrap_fdb_error(env, code, err*)` from util.h to make the error with `api_version_already_set` as the error code (a constant from foundationdb).

**Comment by josephg** [comment]
Great; looks good! Ideally I'd like some unit tests on this (that would make sure this functionality doesn't break in a future version). But for now I'll merge it as-is.

**Comment by josephg** [comment]
This fixes issue #30

**Pull request merged**